br/pkg/lightning/backend/local/local.go

```diff
@@ -1269,76 +965,131 @@ func (local *local) readAndSplitIntoRange(ctx context.Context, engine *Engine, r
 	return ranges, nil
 }
 
-func (local *local) writeAndIngestByRange(
+func (local *local) generateRegionJob(
```

Comment:
`generateRegionJob` sounds like it should return jobs instead of putting jobs into the channel.